code/script.js

```diff
@@ -231,95 +248,169 @@ const setSecret = () => {
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      message = `No, the person doesn't have ${value} ${category}! Remove all people that has ${value} ${category}.`
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
+  showPopUp(message)
+}
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+const showPopUp = (message) => {
+  popUp.style.display = 'block'
+  overlay.style.display = 'block'
+  popUpMessage.innerHTML = message
+}
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+const showPopUpConfirm = (message) => {
+  popUpConfirm.style.display = 'block'
+  overlay.style.display = 'block'
+  confirmMessage.innerHTML = message
+  confirmYesButton.addEventListener('click', checkMyGuess)
+}
 
-  // Invoke a function to redraw the board with the remaining people.
+const closePopUp = () => {
+  popUp.style.display = 'none'
+  overlay.style.display = 'none'
+  popUpConfirm.style.display = 'none'
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+let personToConfirm
+
+// When clicking guess, the player first have to confirm that they want to make a guess.
+const guess = (person) => {
+  personToConfirm = person
+  const messageConfirm = `Are you sure it's ${person}?`
+  showPopUpConfirm(messageConfirm, person)
 }
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+const checkMyGuess = () => {
+
+  personToCheck = personToConfirm
+  if (personToCheck === secret.name) {
+    board.display = 'none'
+    winOrLose.style.display = 'flex'
+    winText.style.display = 'block'
+    loseText.style.display = 'none'
+    secretPersonName.innerHTML = secret.name
+    secretPersonImg.src = secret.img
+  } else {
+    board.display = 'none'
+    winOrLose.style.display = 'flex'
+    loseText.style.display = 'block'
+    winText.style.display = 'none'
+    secretPersonName.innerHTML = secret.name
+    secretPersonImg.src = secret.img
+  }
```

Comment:
Since many actions are the same in the `if` as in the `else`, I think that the `checkMyGuess` function could be shortened like this:
```
const checkMyGuess = () => {
  personToCheck = personToConfirm
  if (personToCheck === secret.name) {
    winText.style.display = 'block'
    loseText.style.display = 'none'
  } else {
    loseText.style.display = 'block'
    winText.style.display = 'none'
  }
    board.display = 'none'
    winOrLose.style.display = 'flex'
    secretPersonName.innerHTML = secret.name
    secretPersonImg.src = secret.img
}
```